packages/x-charts/src/internals/plugins/featurePlugins/useChartPolarAxis/useChartPolarAxis.ts

```diff
@@ -0,0 +1,71 @@
+'use client';
+import * as React from 'react';
+import { warnOnce } from '@mui/x-internals/warning';
+import { ChartPlugin } from '../../models';
+import { ChartSeriesType } from '../../../../models/seriesType/config';
+import { UseChartPolarAxisSignature } from './useChartPolarAxis.types';
+import { useSelector } from '../../../store/useSelector';
+import { selectorChartDrawingArea } from '../../corePlugins/useChartDimensions/useChartDimensions.selectors';
+import { selectorChartSeriesState } from '../../corePlugins/useChartSeries/useChartSeries.selectors';
+import { defaultizeAxis } from './defaultizeAxis';
+
+export const useChartPolarAxis: ChartPlugin<UseChartPolarAxisSignature<ChartSeriesType>> = ({
+  params,
+  store,
+  seriesConfig,
+}) => {
+  const { rotationAxis, radiusAxis, dataset } = params;
+
+  if (process.env.NODE_ENV !== 'production') {
+    const ids = [...(rotationAxis ?? []), ...(radiusAxis ?? [])]
+      .filter((axis) => axis.id)
+      .map((axis) => axis.id);
+    const duplicates = new Set(ids.filter((id, index) => ids.indexOf(id) !== index));
+    if (duplicates.size > 0) {
+      warnOnce(
+        [
+          `MUI X: The following axis ids are duplicated: ${Array.from(duplicates).join(', ')}.`,
+          `Please make sure that each axis has a unique id.`,
+        ].join('\n'),
+        'error',
+      );
+    }
+  }
+
+  const drawingArea = useSelector(store, selectorChartDrawingArea);
+  const formattedSeries = useSelector(store, selectorChartSeriesState);
+
+  // The effect do not track any value defined synchronously during the 1st render by hooks called after `useChartPolarAxis`
+  // As a consequence, the state generated by the 1st run of this useEffect will always be equal to the initialization one
```

Comment:
I didn't understand this comment

Reply:
It's a copy past from the data grid.

We skip the first useEffect, to be in sync with other part of the stor we needs to be sure this hook does not depend on other variable that could be modified by other pluggins running after it.

For charts I don't se a usecases, but for datagrid you might have weird case with the focus, scroll position, and other interaction features.

We can remove the comment if it's confusing.